circuit-lib/circuit-lib.circom/benchmarks/light/utils.ts

```diff
@@ -0,0 +1,335 @@
+        0,
+        1,
+        3,
+        2,
+        2
+    );
+        `;
+    const circuitName = `transaction${nInputUtxos}in${nOutputUtxos}out${
+      this.app ? "App" : "Masp"
+    }MTH${merkleTreeHeight}`;
+
+    this.circuitModels.push({
+      name: circuitName,
+      merkleTreeHeight,
+      nInputUtxos,
+      nOutputUtxos,
+    });
+    const finalPath =
+      this.circuitParameterVariants.outputPath + "/" + circuitName + ".circom";
+
+    // Check if file already exists
+    if (!this.overwrite && existsSync(finalPath)) {
+      return;
+    }
+
+    // Write the generated content to the outputPath
+    writeFileSync(finalPath, circomTemplate.trim());
+  }
+
+  async generateCircomMainFiles() {
+    const {
+      merkleTreeHeightVariants,
+      nInputUtxosVariants,
+      nOutputUtxosVariants,
+    } = this.circuitParameterVariants;
+
+    await this.ensureDirectoryExists(this.circuitParameterVariants.outputPath);
+
+    const parameterVariants = this.combineArrays(
+      merkleTreeHeightVariants,
+      nInputUtxosVariants,
+      nOutputUtxosVariants,
+    );
+
+    for (const variant of parameterVariants) {
+      this.generateCircomMainFile(variant[0], variant[1], variant[2]);
+    }
+  }
+
+  private addPrivateKey(tx: Transaction, account: Account, seed32: string) {
+    const account_privateKey = Account.generateShieldedPrivateKey(
+      seed32,
+      tx.params.poseidon,
+    );
+    const account_publicKey = Account.generateShieldedPublicKey(
+      account_privateKey,
+      tx.params.poseidon,
+    );
+
+    tx.proofInput["inPrivateKey"] = tx.params.inputUtxos.map((utxo: Utxo) => {
+      if (
+        utxo.publicKey.eq(account.pubkey) &&
+        account.pubkey.eq(account_publicKey)
+      ) {
+        return account_privateKey;
+      }
+      if (STANDARD_SHIELDED_PUBLIC_KEY.eq(utxo.publicKey)) {
+        return STANDARD_SHIELDED_PRIVATE_KEY;
+      }
+    });
+  }
+
+  // for now the benchmarks are solely dependant on the Merkle Tree Height
+  async generateProofInputs(merkleTree: MerkleTree) {
+    const lightProvider = await LightProvider.loadMock();
+
+    const seed32 = bs58.encode(new Uint8Array(32).fill(1));
+    const account = new Account({ poseidon: this.poseidon, seed: seed32 });
+    await account.getEddsaPublicKey();
+
+    const depositAmount = 20_000;
+    const depositFeeAmount = 10_000;
+
+    const deposit_utxo1 = new Utxo({
+      poseidon: this.poseidon,
+      assets: [FEE_ASSET, MINT],
+      amounts: [new BN(depositFeeAmount), new BN(depositAmount)],
+      publicKey: account.pubkey,
+      assetLookupTable: lightProvider.lookUpTables.assetLookupTable,
+      verifierProgramLookupTable:
+        lightProvider.lookUpTables.verifierProgramLookupTable,
+    });
+
+    const mockPubkey: PublicKey = SolanaKeypair.generate().publicKey;
+
+    const txParams = new TransactionParameters({
+      outputUtxos: [deposit_utxo1],
```

Comment:
The input utxos are still missing, thus the benchmarks number are probably not representative.
Please add as many input utxos as the respective circuit allows.
https://github.com/Lightprotocol/light-protocol/blob/main/zk.js/tests/transaction.test.ts#L244